### ISSUE TITLE
3.1 Build Broken on NonStop

### ISSUE THREAD

**Issue by rsbeckerca** [openssl/openssl#23221]
As of commit 856d4c6ebc1aec2a714ce497af9c3e48c50e689d  the 3.1 build is now broken on all NonStop platforms based on a requirement that is not satisfied.
```

c99: running /usr/cmplr/ccomfe -Z8 -ZBL --c99 -ZO2 -Zg3 -ZS/tmp/t117440627AHCGAE.T -ZR8 -m -Zf -ZT -D_TANDEM_SOURCE -ZNW203 -ZNW220 -ZNW272 -ZNW734 -ZNW770 -ZNW1506 -Zz -I. -Icrypto -Iinclude -Iproviders/implementations/include -Iproviders/common/include -I/usr/include -DOPENSSL_PIC -DOPENSSLDIR="/usr/local/ssl" -DENGINESDIR="/usr/local/lib64/engines-3" -DMODULESDIR="/usr/local/lib64/ossl-modules" -DOPENSSL_BUILDING_OPENSSL -DOPENSSL_VPROC= -D_XOPEN_SOURCE -D_XOPEN_SOURCE_EXTENDED=1 -D_TANDEM_SOURCE -DB_ENDIAN -DNDEBUG providers/implementations/ciphers/cipher_sm4_ccm.c 
  IMPLEMENT_aead_cipher(sm4, ccm, CCM, AEAD_FLAGS, 128, 8, 96);
  ^
"/home/jenkinsbuild/.jenkins/workspace/OpenSSL-3.1_Monitor/providers/implementations/ciphers/cipher_sm4_ccm.c", line 39: error(114):
          identifier "sm4_ccm_dupctx" is undefined
```
No change has been made to the build configuration at 3.1.

**Comment by t8m** [comment]
This is fixed now with #23217 merged.

**Comment by rsbeckerca** [comment]
Thanks @nhorman. The main build passes now.
